File: Stage15/Python/Problem4948.py
import math

def isPrime(n):
    if n < 2:
        return False
    if n == 2:
        return True
    if n % 2 == 0:
        return False
    
    for i in range(3, int(math.sqrt(n)) + 1, 2):
        if n % i == 0:
            return False
    return True

def main():
    while (True):
        n = int(input())
        if (n == 0):
            break
        count = 0
        
        for i in range(n + 1, 2 * n + 1):
            if (isPrime(i)):
                count += 1
        print(count)
        
    
if __name__ == "__main__":
    main()

File: Stage15/C/Problem4948.c
#include <stdio.h>
#include <stdbool.h>

bool isPrime(int n) {
    if (n < 2) return false;
    if (n == 2) return true;
    if (n % 2 == 0) return false;

    for (int i = 3; i * i <= n; i += 2) {
        if (n % i == 0) return false;
    }
    return true;
}

int main(void) {
    int n;

    while (1) {
        scanf("%d", &n);
        if (n == 0) break;
        int count = 0;

        for (int i = n + 1; i <= 2 * n; i++) {
            if (isPrime(i)) count++;
        }
        printf("%d\n", count);
    }

    return 0;
}